### PR TITLE
SW-3903 Add Planting Site Selector for Map View

### PR DESCRIPTION
**Pull request by charlie-tf**
- create a common PlantingSiteSelector component
- create a component that variably displays search or the selector based on view
- keep track of the selected planting site id and pass to map

image

### CODE REVIEW

File: src/components/NurseryWithdrawals/PlantingProgressTabContent.tsx [hunk 1]
```diff
@@ -58,11 +64,31 @@ export default function PlantingProgress(): JSX.Element {
...
     </Card>
   );
 }
+
+type SearchComponentProps = SearchProps & {
+  view: View;
+  onChangePlantingSite: (newSiteId: number) => void;
+};
+
+function SearchComponent(props: SearchComponentProps): JSX.Element {
```
[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/components/NurseryWithdrawals/PlantingProgressTabContent.tsx:82]: This is great!!

File: src/components/common/PlantingSiteSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { selectPlantingSites } from 'src/redux/features/tracking/trackingSelectors';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import strings from 'src/strings';
+
+type PlantingSiteSelectorProps = {
+  onChange: (plantingSiteId: number) => void;
+};
+
+export default function PlantingSiteSelector({ onChange }: PlantingSiteSelectorProps): JSX.Element {
+  // assume `requestPlantingSites` thunk has been dispatched by consumer
```
Comment (karthikbtf) [inline src/components/common/PlantingSiteSelector.tsx:12]: plantings sites request is dispatched in App.tsx

File: src/components/common/PlantingSiteSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    [plantingSites]
+  );
+
+  const updateSelection = useCallback(
+    (newValue: any) => {
+      const id = Number(newValue);
+      setSelectedPlantingSiteId(isNaN(id) ? -1 : id);
+      onChange(isNaN(id) ? -1 : id);
+    },
+    [onChange, setSelectedPlantingSiteId]
```
Comment (karthikbtf) [inline src/components/common/PlantingSiteSelector.tsx:27]: do we need the setSelectedPlantingSiteId?

**Review by karthikbtf — APPROVED** [review]
Nice 🎉

File: src/components/common/PlantingSiteSelector.tsx (hunk 3 above)
Comment (charlie-tf) [reply src/components/common/PlantingSiteSelector.tsx:27]: I thought so—I'll try deleting and see if the linter complains...

Comment (charlie-tf) [reply src/components/common/PlantingSiteSelector.tsx]: Nope! not needed

Comment (karthikbtf) [reply src/components/common/PlantingSiteSelector.tsx]: Thanks for checking \o/

**Review by karthikbtf — APPROVED** [review]

**Pull request merged**